javav2/example_code/opensearch/src/main/java/com/example/search/scenario/OpenSearchScenario.java

```diff
@@ -0,0 +1,323 @@
+
+    static OpenSearchActions openSearchActions = new OpenSearchActions();
+
+    public static void main(String[] args) throws Throwable {
+        logger.info("""
+            Welcome to the Amazon OpenSearch Service Basics Scenario.
+
+            Use the Amazon OpenSearch Service API to create, configure, and manage OpenSearch Service domains.
+
+            The operations exposed by the AWS OpenSearch Service client are focused on managing the OpenSearch Service domains 
+            and their configurations, not the data within the domains (such as indexing or querying documents). 
+            For document management, you typically interact directly with the OpenSearch REST API or use other libraries, 
+            such as the OpenSearch Java client (https://opensearch.org/docs/latest/clients/java/).
+
+            Let's get started...
+        """);
+        waitForInputToContinue(scanner);
+        try {
+            runScenario();
+        } catch (RuntimeException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private static void waitForInputToContinue(Scanner scanner) {
+        while (true) {
+            logger.info("");
+            logger.info("Enter 'c' followed by <ENTER> to continue:");
+            String input = scanner.nextLine();
+
+            if (input.trim().equalsIgnoreCase("c")) {
+                logger.info("Continuing with the program...");
+                logger.info("");
+                break;
+            } else {
+                logger.info("Invalid input. Please try again.");
+            }
+        }
+    }
+
+    private static void runScenario() throws Throwable {
+        String currentTimestamp = String.valueOf(System.currentTimeMillis());
+        String domainName = "test-domain-" + currentTimestamp;
+
+        logger.info(DASHES);
+        logger.info("1. Create an Amazon OpenSearch domain");
+        logger.info("""
+            An Amazon OpenSearch domain is a managed instance of the OpenSearch engine, 
+            which is an open-source search and analytics engine derived from Elasticsearch. 
+            An OpenSearch domain is essentially a cluster of compute resources and storage that hosts 
+            one or more OpenSearch indexes, enabling you to perform full-text searches, data analysis, and 
+            visualizations.
+
+            In this step, we'll initiate the creation of the domain. We'll check on the progress in a later step.
+        """);
+        waitForInputToContinue(scanner);
+
+        try {
+            CompletableFuture<String> future = openSearchActions.createNewDomainAsync(domainName);
+            String domainId = future.join();
+            logger.info("Domain successfully created with ID: {}", domainId);
+        } catch (RuntimeException rt) {
+            Throwable cause = rt.getCause();
+            if (cause != null) {
+                if (cause instanceof OpenSearchException openSearchEx) {
```

Comment:
Why do you use "if" here, and use "while" in other places to find the specific exception?
See line 127.